### PR TITLE
KAFKA-10092: Remove unused code branches in NioEchoServer

### PR DESCRIPTION
**Pull request by afalko**
NioEchoServer the enum has its constructor declared as private, which is redundant. There is also an unused exception throws and unused method.

Reviewers: Jakob Homan <jghoman@gmail.com>

This is a newbie ticket to get used to the contribution flow.

### CODE REVIEW

**Comment by afalko** [comment]
@jghoman

**Review by jghoman — APPROVED** [review]

File: clients/src/test/java/org/apache/kafka/common/network/NioEchoServer.java [hunk 1]
```diff
@@ -60,14 +60,8 @@
...
 
-        private final String metricNameSuffix;
-
-        private MetricType() {
-            metricNameSuffix = "-" + name().toLowerCase(Locale.ROOT);
-        }
-
         public String metricNameSuffix() {
-            return metricNameSuffix;
+            return "-" + name().toLowerCase(Locale.ROOT);
```
Comment (ijuma) [inline clients/src/test/java/org/apache/kafka/common/network/NioEchoServer.java]: This replaces a single allocation during construction with allocations on every method call.

**Review by ijuma — COMMENTED** [review]
Thanks for the PR. Not sure we want to make this change, see below.

**Comment by jghoman** [comment]
ok to test

**Review by andrewchoi5 — APPROVED** [review]

**Review by notifygd — APPROVED** [review]

**Comment by chia7712** [comment]
@afalko Could you rebase PR to trigger QA?

**Review by chia7712 — APPROVED** [review]

**Comment by afalko** [comment]
> @afalko Could you rebase PR to trigger QA?

Thanks! Looks like it partially passed. I'll merge master against and see if it does better.

**Comment by chia7712** [comment]
@afalko Thanks for updating code. Are those failed tests related to this patch?

**Comment by afalko** [comment]
> @afalko Could you rebase PR to trigger QA?

Thanks @chia7712 

> @afalko Thanks for updating code. Are those failed tests related to this patch?

@chia7712  These are not, last time it ran the jdk8 and jdk11 tests passed, but the jdk15 tests failed, this time it is the other way around.

**Comment by afalko** [comment]
I'll merge master once more

**Review by chia7712 — APPROVED** [review]
@afalko Thanks for your patch. LGTM

**Pull request merged**